embedded/store/ongoing_tx.go

```diff
@@ -215,16 +234,41 @@ func (tx *OngoingTx) AddPrecondition(c Precondition) error {
 	return nil
 }
 
-func (tx *OngoingTx) ExistKeyWith(prefix, neq []byte) (bool, error) {
+func (tx *OngoingTx) GetWithPrefix(prefix, neq []byte) (key []byte, valRef ValueRef, err error) {
 	if tx.closed {
-		return false, ErrAlreadyClosed
+		return nil, nil, ErrAlreadyClosed
 	}
 
 	if tx.IsWriteOnly() {
-		return false, ErrWriteOnlyTx
+		return nil, nil, ErrWriteOnlyTx
 	}
 
-	return tx.snap.ExistKeyWith(prefix, neq)
+	key, valRef, err = tx.snap.GetWithPrefix(prefix, neq)
+	if errors.Is(err, ErrKeyNotFound) {
+		expectedGetWith := expectedGetWithPrefix{
+			prefix: cp(prefix),
+			neq:    cp(neq),
+		}
+
+		tx.expectedGetsWithPrefix = append(tx.expectedGetsWithPrefix, expectedGetWith)
```

Comment:
is `expectedGetsWithPrefix` used to filter keys which are not found? In `checkPreconditions` I see the check is being made against a latest snapshot, just wanted to understand the flow (if possible by adding a comment on what `expectedGetsWithPrefix/filters ` does)